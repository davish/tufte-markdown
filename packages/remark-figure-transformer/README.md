### `@tufte-markdown/remark-figure-transformer`

The figure-transformer
