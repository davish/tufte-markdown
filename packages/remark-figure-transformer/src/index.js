export { default } from './remark-figure-transformer'
