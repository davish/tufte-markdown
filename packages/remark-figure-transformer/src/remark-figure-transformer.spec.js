import u from 'unist-builder'
import { transformer } from './remark-figure-transformer'

describe('with text-content', () => {
  let ast
  let result
  beforeEach(() => {
    ast = u('root', [
      u(
        'tufteFigure',
        {
          attributes: {
            openingHtml: '<figure>',
            closingHtml: '</figure>',
          },
        },
        [u('text', 'This is a text content within a parsed tufte figure')]
      ),
    ])

    result = transformer(ast).children[0]
  })

  it('should match snapshot', () => {
    expect(result).toMatchSnapshot()
  })

  it('should output a `paragraph`-node', () => {
    expect(result.type).toEqual('paragraph')
  })

  it('should contain the hints that the paragraph is rendered as `figure`', () => {
    expect(result.data.hName).toEqual('figure')
  })

  it('should pass on the children', () => {
    expect(result.children).toHaveLength(1)
    expect(result.children[0].type).toEqual('text')
  })
})

describe('with markdown-content', () => {
  let ast
  let result
  beforeEach(() => {
    ast = u('root', [
      u(
        'tufteFigure',
        {
          attributes: {
            openingHtml: '<figure>',
            closingHtml: '</figure>',
          },
        },
        [
          u('text', 'This is a text content within a parsed tufte figure'),
          u('image', { url: 'http://example.org/foo.png' }),
        ]
      ),
    ])

    result = transformer(ast).children[0]
  })

  it('should match snapshot', () => {
    expect(result).toMatchSnapshot()
  })
})

describe('with additional classes', () => {
  let ast
  let result
  beforeEach(() => {
    ast = u('root', [
      u(
        'tufteFigure',
        {
          attributes: {
            openingHtml: '<figure class="fullwidth">',
            closingHtml: '</figure>',
          },
        },
        [
          u('text', 'This is a text content within a parsed tufte figure'),
          u('image', { url: 'http://example.org/foo.png' }),
        ]
      ),
    ])

    result = transformer(ast).children[0]
  })

  it('should match snapshot', () => {
    expect(result).toMatchSnapshot()
  })

  it('should pass on classes to `paragraph`-node', () => {
    expect(result.data.hProperties.className).toEqual('fullwidth')
  })
})
