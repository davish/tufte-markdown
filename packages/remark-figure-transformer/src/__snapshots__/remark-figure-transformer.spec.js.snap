// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with additional classes should match snapshot 1`] = `
Object {
  "children": Array [
    Object {
      "type": "text",
      "value": "This is a text content within a parsed tufte figure",
    },
    Object {
      "type": "image",
      "url": "http://example.org/foo.png",
    },
  ],
  "data": Object {
    "hName": "figure",
    "hProperties": Object {
      "className": "fullwidth",
    },
  },
  "type": "paragraph",
}
`;

exports[`with markdown-content should match snapshot 1`] = `
Object {
  "children": Array [
    Object {
      "type": "text",
      "value": "This is a text content within a parsed tufte figure",
    },
    Object {
      "type": "image",
      "url": "http://example.org/foo.png",
    },
  ],
  "data": Object {
    "hName": "figure",
    "hProperties": undefined,
  },
  "type": "paragraph",
}
`;

exports[`with text-content should match snapshot 1`] = `
Object {
  "children": Array [
    Object {
      "type": "text",
      "value": "This is a text content within a parsed tufte figure",
    },
  ],
  "data": Object {
    "hName": "figure",
    "hProperties": undefined,
  },
  "type": "paragraph",
}
`;
